src/System.Management.Automation/engine/ParameterBinderController.cs

```diff
@@ -260,6 +260,32 @@ internal void ReparseUnboundArguments()
             UnboundArguments = result;
         }
 
+        protected void InitUnboundArguments(Collection<CommandParameterInternal> arguments)
+        {
+            // Add the passed in arguments to the unboundArguments collection
+            Collection<CommandParameterInternal> paramsFromSplatting = null;
+            foreach (CommandParameterInternal argument in arguments)
+            {
+                if (argument.FromHashtableSplatting)
+                {
+                    paramsFromSplatting ??= new Collection<CommandParameterInternal>();
+                    paramsFromSplatting.Add(argument);
+                }
+                else
+                {
+                    UnboundArguments.Add(argument);
+                }
+            }
+
+            if (paramsFromSplatting != null)
```

Comment:
Please add comment as to why parameters from splatting are pushed to the end of the UnboundArguments list.

Reply:
Comments added.